Golem.Tests/Utils.cs

```diff
@@ -110,9 +112,52 @@ public async static Task<T> ReadChannel<T>(ChannelReader<T> channel, Func<T, boo
 
             throw new Exception($"Failed to find matching {nameof(T)} within {timeoutMs} ms.");
         }
+
+        public static void CheckYagnaApiPortIsFree()
+        {
+            if (!IsPortFree(EnvironmentBuilder.DefaultAddress, EnvironmentBuilder.DefaultApiPort))
+            {
+                throw new Exception("API port is open");
+            }
+        }
+
+        public static bool IsPortFree(
+            string host,
+            int port,
+            int retries = 3,
+            int retry_delay = 500,
+            int connection_timeout = 250
+        )
+        {
+            for (int retries_count = 0; retries_count < retries; retries_count++)
+            {
+                try
+                {
+                    var client = new TcpClient();
+                    IAsyncResult connection = client.BeginConnect(host, port, null, null);
+                    if (!connection.AsyncWaitHandle.WaitOne(connection_timeout))
+                        return true;
+                    client.EndConnect(connection);
+                }
+                catch (Exception)
+                {
+                    return true;
+                }
+                Thread.Sleep(retry_delay);
+            }
+            return false;
+        }
+    }
+
+    public class WithFreeApiPort
+    {
+        public WithFreeApiPort()
+        {
+            TestUtils.CheckYagnaApiPortIsFree();
```

Comment:
Constructor of parent class fails if some remaining process holds open API port.